### ISSUE TITLE
Multiteam collaboration support

### ISSUE THREAD

**Issue by maryoush** [argoproj/argo#2042]
# Summary

Was wondering about potential capabilities or mitigations of ArgoWorkflow in order to support that multi-team can work on single workflow.  The most aspects are security where minimal visibility of steps between teams/namespaces need to be kept.
 
* Each team can provide own step in separate namespace
* Each team can test its step separately (in own namespace)
* Each step can use dedicated credentials (e.g for backing service being called)
no other team can use that /see that credentials.
* Workflow owner  (namespace), account has limited access to get each'es steps dedicated credentials/configuration.

Reading some lines of your document that shows that step per name space deployment is not supported

Also multi service accounts is not possible for each step. 
```
All pods in a workflow run with the service account specified in workflow.spec.serviceAccountName,
or if omitted, the default service account of the workflow's namespace 
```
Is this anything on your backlog addressing that scenario ?

**Comment by maryoush** [comment]
Are there any plans to support cross namespace deployment for workflow steps ?

**Comment by simster7** [comment]
We are having abstract talks on the allowing this behavior, but haven't prioritized it yet. Regardless, I definitely wouldn't expect this to be available before `2.6`.

**Comment by alexec** [comment]
See #3523
